frontend/src/components/Caption.js

```diff
@@ -15,46 +16,29 @@ class Caption extends React.Component {
         super(props);
         this.state = {
             caption: props.caption, 
-            isHighlighted: false,
+            isHighlighted: props.scrollTo === props.caption.caption_id,
             token: props.token,
             showAuthModal: false,
         }
     }
     
-    componentDidMount = () => this.scrollToCaption();
-
-    componentDidUpdate = () => {
-        const captionId = this.state.caption.caption_id;
-        const captionRef = this.refs[captionId];
-        this.highlightCaption(captionRef);
-    }
-
-    onBlur = (event) => {
-        const { scrollTo } = this.props;
-        const captionId = this.state.caption.caption_id;
-        if ((scrollTo === captionId) && scrollTo) {
-            this.setState({ isHighlighted: false });
-        }
-    }
+    componentDidMount = () => {
+        this.scrollToCaption()
```

Comment:
It makes more sense to have the `isHighlighted` check here rather than in `scrollToCaption`